validator_client/src/block_service.rs

```diff
@@ -852,6 +845,43 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
 
         Ok::<_, BlockError>(unsigned_block)
     }
+
+    /// translate `builder_proposals``, `builder_boost_factor`` and `prefer_builder_proposals`` to a
+    /// boost factor. if `builder_proposals` is set to false, set boost factor to 0 to indicate a
+    /// preference for local payloads. if builder_boost_factor is a value other than none, return
+    /// its value as the boost factor. if `prefer_builder_proposals` is true, set boost factor to
+    /// u64::MAX to indicate a preference for builder payloads. else return None to indicate no preference
+    /// between builder and local payloads
+    fn get_builder_boost_factor(&self, validator_pubkey: &PublicKeyBytes) -> Option<u64> {
+        let builder_proposals = self.validator_store.get_builder_proposals(validator_pubkey);
+
+        let builder_boost_factor = self
+            .validator_store
+            .get_builder_boost_factor(validator_pubkey);
+
+        let prefer_builder_proposals = self
+            .validator_store
+            .get_prefer_builder_proposals(validator_pubkey);
+
+        if !builder_proposals {
+            return Some(0);
+        }
+
+        if let Some(builder_boost_factor) = builder_boost_factor {
+            // if builder boost factor is set to 100 it should be treated
+            // as None to prevent unnecessary calculations that could
+            // lead to loss of information.
+            if builder_boost_factor != 100 {
+                return Some(builder_boost_factor);
+            }
+        }
+
+        if prefer_builder_proposals {
+            return Some(u64::MAX);
+        }
```

Comment:
```suggestion
        if prefer_builder_proposals {
            return Some(u64::MAX);
        }

        if let Some(builder_boost_factor) = builder_boost_factor {
            // if builder boost factor is set to 100 it should be treated
            // as None to prevent unnecessary calculations that could
            // lead to loss of information.
            if builder_boost_factor != 100 {
                return Some(builder_boost_factor);
            }
        }
```

I'm not sure if there's an objective truth behind how these preferences should work, but I was thinking that it might be easier for debugging this way.

I imagine a confused config that has `prefer_builder_payloads == True` and `builder_boost_factor == 0`. I feel like it would be easier to answer the question "why is this still preferring builder payloads?" rather than "why is it preferring local payloads?".